src/main/features/claim/helpers/feesTableViewHelper.ts

```diff
@@ -36,47 +36,54 @@ class RangeUtils {
    * @returns {boolean} true if ranges overlap, otherwise false
    */
   static areOverlap (lhs: RangePartial, rhs: RangePartial): boolean {
-    return (lhs.from <= rhs.from && lhs.to >= rhs.from) || (lhs.from <= rhs.to && lhs.to >= rhs.to) || (lhs.from <= rhs.from && lhs.to >= rhs.to)
+    return (lhs.minRange <= rhs.minRange && lhs.maxRange >= rhs.minRange) || (lhs.minRange <= rhs.maxRange && lhs.maxRange >= rhs.maxRange) || (lhs.minRange <= rhs.minRange && lhs.maxRange >= rhs.maxRange)
   }
+
 }
 
 class Item {
-  constructor (public readonly range: Range, public readonly targetColumn: number) {}
+  constructor (public readonly range: FeeRange, public readonly targetColumn: number) {}
 
-  static createForFeeInColumn (range: Range, targetColumn: number): Item {
+  static createForFeeInColumn (range: FeeRange, targetColumn: number): Item {
     return new Item(range, targetColumn)
   }
 }
 
-export class Row {
-  constructor (public readonly from: number, public readonly to: number, private readonly fees: { [key: number]: number } = {}) {}
+export class FeeRange implements RangePartial {
+  constructor (public readonly minRange: number, public readonly maxRange: number, public readonly amount: number | string) {}
+}
+
+export class FeeRangeMerge implements RangePartial {
+  constructor (public readonly minRange: number, public readonly maxRange: number, private readonly fees: { [key: string]: number | string } = {}) {}
 
-  addFee (key: number, value: number): void {
+  addFee (key: number, value: number | string): void {
     this.fees[key] = value
   }
 }
 
 export class FeesTableViewHelper {
-  static merge (firstFeesSet: Range[], secondFeesSet: Range[], increment: number = 1): Row[] {
+  static merge (firstFeesSet: FeeRange[], secondFeesSet: FeeRange[], increment: number = 1): FeeRangeMerge[] {
+    // console.log(firstFeesSet, secondFeesSet)
     if (firstFeesSet === undefined || secondFeesSet === undefined) {
       throw new Error('Both fee sets are required for merge')
     }
-
     const items: Item[] = [
         ...firstFeesSet.map(range => Item.createForFeeInColumn(range, 1)),
       ...secondFeesSet.map(range => Item.createForFeeInColumn(range, 2))
     ].sort((lhs: Item, rhs: Item) => RangeUtils.compare(lhs.range, rhs.range))
 
-    return items.reduce((rows: Row[], item: Item) => {
-      const overlappedRows: Row[] = rows.filter((row: Row) => RangeUtils.areOverlap(item.range, row))
-
+    return items.reduce((rows: FeeRangeMerge[], item: Item) => {
```

Comment:
We should rename variable name as well. I forgotten to do it when I renamed class. The same in other places which are not in sync.